libs/frontend/feature-site-assets/src/components/SelectAssetModal.vue

```diff
@@ -0,0 +1,288 @@
+    </div>
+  </Modal>
+</template>
+
+<script lang="ts" setup>
+import { computed, inject, onMounted, ref, toRefs, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+import {
+  Check,
+  Modal,
+  PSButton,
+  PSInput,
+  PSMultiselect,
+  PSSpinner,
+} from '@pubstudio/frontend/ui-widgets'
+import { ISiteViewModel } from '@pubstudio/shared/type-api-platform-site'
+import { useSites } from '@pubstudio/frontend/feature-sites'
+import { store } from '@pubstudio/frontend/data-access-web-store'
+import {
+  AssetContentType,
+  SiteAssetState,
+  IListPlatformSiteAssetsRequest,
+  ISiteAssetViewModel,
+} from '@pubstudio/shared/type-api-platform-site-asset'
+import { debounce } from '@pubstudio/shared/util-debounce'
+import { ApiInjectionKey } from '@pubstudio/frontend/data-access-injection'
+import { PSApi } from '@pubstudio/frontend/util-api'
+import { useSiteAssetApi } from '@pubstudio/frontend/data-access-api'
+
+const { t } = useI18n()
+
+const props = withDefaults(
+  defineProps<{
+    show: boolean
+    initialSiteId?: string
+    contentTypes?: AssetContentType[]
+  }>(),
+  {
+    initialSiteId: undefined,
+    contentTypes: undefined,
+  },
+)
+const { show, initialSiteId } = toRefs(props)
+
+const emit = defineEmits<{
+  (e: 'cancel'): void
+  (e: 'select', asset: ISiteAssetViewModel): void
+}>()
+
+const rootApi = inject(ApiInjectionKey) as PSApi
+const api = useSiteAssetApi(rootApi)
+
+const { sites, listSites } = useSites()
+
+const defaultFilter = (): IListPlatformSiteAssetsRequest => {
+  let siteId = initialSiteId.value
+
+  if (siteId === 'scratch') {
+    siteId = undefined
+  } else if (siteId === 'identity') {
+    siteId = store.user.identity.value.id
+  }
+
+  return {
+    site_id: siteId,
+    content_type: undefined,
+    search: undefined,
+  }
+}
+
+const loading = ref(false)
+const siteAssets = ref<ISiteAssetViewModel[]>()
+const filter = ref<IListPlatformSiteAssetsRequest>(defaultFilter())
+const selectedAsset = ref<ISiteAssetViewModel>()
+
+const resolvedSites = computed(() => [store.user.identity.value, ...(sites.value ?? [])])
+
+const updateSite = async (site: ISiteViewModel | undefined) => {
+  filter.value.site_id = site?.id
+  await listAssets()
+}
+
+const updateSearch = debounce(async (search: string) => {
+  filter.value.search = search
+  await listAssets()
+}, 500)
+
+const updateContentType = async (contentType: AssetContentType | undefined) => {
+  filter.value.content_type = contentType
+  await listAssets()
+}
+
+const cancel = () => {
+  cleanup()
+  emit('cancel')
+}
+
+const cleanup = () => {
+  filter.value = defaultFilter()
+  selectedAsset.value = undefined
+}
+
+const emitSelect = () => {
+  if (selectedAsset.value) {
+    emit('select', selectedAsset.value)
+  }
+}
+
+const listAssets = async () => {
+  loading.value = true
+  try {
+    const response = await api.listSiteAssets({
+      ...filter.value,
+      user_id: store.user.id.value,
+      state: SiteAssetState.Uploaded,
```

Comment:
The modal won't show any assets in development mode because there's no site assets in `Uploaded` state in the fixture. You can comment out this line (182) to temporarily see what does it look like when there's asset in the modal.

![Screenshot 2023-11-23 at 4 21 42 PM](https://github.com/samatechtw/pubstudio-builder/assets/59676941/87f6f09d-f13b-4010-ae4a-63a395139d54)